runware/utils.py

```diff
@@ -767,3 +771,35 @@ def instantiateDataclassList(dataclass_type: Type[Any], data_list: List[dict]) -
     for data in data_list:
         instances.append(instantiateDataclass(dataclass_type, data))
     return instances
+
+
+def get_cn_params_and_uuid_model(control_data) -> Tuple[Dict[str, Any], Type[IControlNetWithUUID]]:
+    """
+
+    Args:
+        control_data: dict: A dictionary with controlnet data
+
+    Returns:
+        A dict with preprocessing params
+        An UUID Model to use ControlNet
+    """
+    uuid_model = IControlNetGeneralWithUUID
+    params = {}
+    if hasattr(control_data, "preprocessor"):
+        uuid_model = IControlNetAWithUUID
+        params.update({
+            "preprocessor": control_data.preprocessor.value,
+            "guideImageUnprocessed": control_data.guideImageUnprocessed
+        })
+        if isinstance(control_data, IControlNetCanny):
+            uuid_model = IControlNetCannyWithUUID
+            params.update({
+                "lowThresholdCanny": control_data.lowThresholdCanny,
+                "highThresholdCanny": control_data.highThresholdCanny,
+            })
+        elif isinstance(control_data, IControlNetHandsAndFace):
+            uuid_model = IControlNetHandsAndFaceWithUUID
```

Comment:
what is this? uuid + model? it's like mixing two different concepts, so I'm confused. also why is needed all this "WithUUID" thing?